npm-pack-and-replace.sh

```diff
@@ -50,7 +50,11 @@ replace_in_pwd_package_json() {
   echo "Packing and replacing $package..."
 
   # pack the package
-  tar_file="$( (cd "$path" && npm run build && npm pack --pack-destination="$current_dir") | tail -n 1)"
+  pushd "$path"
+  npm run build
+  pack_out=$(npm pack --pack-destination="$current_dir")
+  tar_file=$(tail -n 1 <<<"$pack_out")
+  popd
```

Comment:
to be honest I like the subconsole approach more, because here once npm run build exits with error, the pwd in current terminal remains changed. If something fails inside subconsole `( ... && ... )`, it doesn't change the pwd

Reply:
What terminal are you using and how are you running it? I would expect it to leave your cwd alone no matter what the script does. I don't seem to have that issue when I run `../../npm-pack-and-replace.sh` in zsh, my terminal is still in `chain-cli/chaincode-template` even if `npm run build` fails there.

Reply:
interesting, after verifying that it works as expected (the path remains as it should be). So sorry for that, you may ignore my prev comment